include/openenclave/bits/eeid.h

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file eeid.h
+ *
+ * This file defines the EEID structure.
+ *
+ */
+
+#ifndef _OE_BITS_EEID_H
+#define _OE_BITS_EEID_H
+
+#include "properties.h"
+
+OE_EXTERNC_BEGIN
+
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+
+// This is the public key corresponding to the private key OE_DEBUG_SIGN_KEY in
+// signkey.c/.h.
+static const uint8_t OE_DEBUG_PUBLIC_KEY[] = {
+    0xca, 0x9a, 0xd7, 0x33, 0x14, 0x48, 0x98, 0x0a, 0xa2, 0x88, 0x90,
+    0xce, 0x73, 0xe4, 0x33, 0x63, 0x83, 0x77, 0xf1, 0x79, 0xab, 0x44,
+    0x56, 0xb2, 0xfe, 0x23, 0x71, 0x93, 0x19, 0x3a, 0x8d, 0xa};
+
+/**
+ * Determine whether properties are those of a base image to be used with EEID
+ *
+ * @param[in] properties Properties of an SGX enclave.
+ *
+ * @returns Returns true if **properties** are consistent with those of an EEID
+ * base image.
+ *
+ */
+int is_eeid_base_image(const oe_sgx_enclave_properties_t* properties);
+
+/*
+**==============================================================================
+**
+** oe_eeid_t
+**
+**==============================================================================
+*/
+
+#define OE_EEID_VERSION (1)
+
+OE_PACK_BEGIN
```

Comment:
Sorry, I should have caught this earlier, but it only just occurred to me that these are being released in a _public_ header; could you do me a favor and refactor out the actual publicly required pieces (i.e. something an end user developer would need) vs. the internal ones (i.e. functions shared between host/enclave or attester/verifier plugin). We would like to be very scrupulous in tracking these abstraction boundaries so that devs don't end up taking unintended dependencies on functions we don't expect. 

As far as I can tell, only the `oe_eeid_t` data structure needs to be in bits and the rest should be part of a new `include/openenclave/internal/sgx/eeid.h` header (or some of the methods like the new ntoh/hton ones can go into the `include/openenclave/internal/sgx/eeid_plugin.h` as you feel appropriate)

Reply:
Sure, no problem!

Reply:
Done!